### PR TITLE
feat(s3): update runtime of notifications-handler to python3.9 

### PR DESCRIPTION
**Pull request by watany-dev**
closes https://github.com/aws/aws-cdk/issues/20973

This upgrades the runtime of the internal function `notifications-resource-handler` from Python 3.7 to Python 3.9

Referring to the PR below
https://github.com/aws/aws-cdk/pull/21483/

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by watany-dev** [comment]
I have updated the existing integ test and verified that it is the updated runtime. Do I need to create any more new tests?

**Comment by mrgrain** [comment]
Code looks good. I checked and python3.9 is available everywhere we need it.

Once the test passes, this should be good to ship.

**Comment by watany-dev** [comment]
Glad to have your review.
I checked and it looks like the integ test needed @s3-notification as well as @s3. I'm currently waiting for the results of a push that resolves this.

**Comment by watany-dev** [comment]
@mrgrain 
We were able to confirm the PASS of all tests.

File: packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { md5hash } from '@aws-cdk/core/lib/helpers-internal';
 import { ISecurityGroup, IVpc, SubnetSelection } from '@aws-cdk/aws-ec2';
 import * as iam from '@aws-cdk/aws-iam';
 import * as lambda from '@aws-cdk/aws-lambda';
 import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
 import { Stack, Names } from '@aws-cdk/core';
+import { md5hash } from '@aws-cdk/core/lib/helpers-internal';
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts:6]: Would you mind reverting this unrelated change. Thanks.

**Review by mrgrain — CHANGES_REQUESTED** [review]
Just a minor thing.

But I'll need some more time to look at these integ tests. I'm a bit surprised so many snapshots need updating.

File: packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts (hunk 1 above)
Comment (mrgrain) [reply packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts]: Ah I see, thanks for explaining where this came from. 👍🏻 Probably would have been fine to leave it in then. From a PR review it's difficult to establish the source of these kind of changes (and manually checking would just take to long).

Comment (watany-dev) [reply packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts:6]: Sorry I was in a hurry to fix it. This commit should bring us back to the beginning of this review!

Comment (mrgrain) [reply packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts:6]: All good and thank you for being diligent. It's a tricky thing. When the same module has other change I usually don't question it.

Maybe adding a comment to the line on GitHub would help reviewers. 🤔

**Comment by mrgrain** [comment]
This looks all good. I'm currently looking into adding an extended test that actually tests the notification delivery. Not sure if it's possible though.

**Review by mrgrain — APPROVED** [review]
🚀
